matrix_sdk/src/request_builder.rs

```diff
@@ -270,6 +281,7 @@ impl Into<get_message_events::Request> for MessagesRequestBuilder {
             from: self.from,
             to: self.to,
             dir: self.direction.unwrap_or(Direction::Backward),
+            // Will our default overwrite the normal default
             limit: self.limit.map(UInt::from).unwrap_or_default(),
```

Comment:
If this is how [deserialization](https://github.com/ruma/ruma/blob/master/ruma-client-api/src/r0/message/get_message_events.rs#L51) is handled will the server ignore a `limit = 0`

Reply:
We really need constructors for Ruma types badly, especially where we're applying defaults to fields in `Deserialize` that aren't the types `Default`.

Reply:
I've added constructors for the request types in `create_room` and `get_message_events`.

Reply:
I still have to work on using the constructors.